### ISSUE TITLE
[Request] run-block should support storage import

### ISSUE THREAD

**Issue by crystalin** [AcalaNetwork/chopsticks#225]
Currently it seems that using the import storage is not compatible with `run-block`. 
I briefly looked at it and it seems to be because the import storage is actually applying the change on top of a block which might be incompatible with `run-block`. 

Not sure if you have an idea on how to support it before I dig into it

**Comment by ermalkaleci** [comment]
It's supposed to "replay" a block. Why do you need to override storage?

**Comment by crystalin** [comment]
I agree it is not very intuitive, but for the same as to why you want to override a wasm runtime, it is sometime important to see what would have happened to a block if your runtime or storage was different.

In my specific case, I want to verify if a given smart contract (stored in the chain storage) could have worked properly if a given instruction was changed in its code.

**Comment by ermalkaleci** [comment]
I see, yeah it's easy to implement it

**Comment by ermalkaleci** [comment]
Btw we also have dry-run command which can be run at any block

**Comment by crystalin** [comment]
Yes, that is convenient for single extrinsic, but sometime it is a combination of multiple factors included in a full block.
I was able to test what I wanted by modifying the import storage to include the blockhash, but it was hardcoded. I'll check if I can propose something more generic

**Comment by crystalin** [comment]
Awesome, thank you :)
